### ISSUE TITLE
Some questions about restoration design and the restoration factor

### ISSUE THREAD

**Issue by kadmkbl** [qinenergy/cotta#5]
Hi, author! Thanks again for your sharing codes. I've fixed the problems occurring in reproducing CIFAR-10C results by replacing the model back to the standard one. And I am here for some questions in your proposed restoration design.
We know that "Stochastic Restoration" in CoTTA is designed to avoid catastrophic forgetting. The thing that bothers me is, in CL, avoiding catastrophic forgetting is needed when you want a model to solve all the tasks/domains that have been seen previously. But when all you need is just to keep high performance in current tasks/domains (i.e., online performance), what might be the reason for choosing restoration to avoid catastrophic forgetting (i.e., keep previous knowledge)? I know it achieves better results, and as the paper mentioned, it helps recovery after encountering hard examples. And I'll still appreciate it very much if there is any answer for my puzzles (it is also totally okay not to do so :) ).
Or maybe is there any advice for the restoration factor setup (why CoTTA sets 0.01) ? :)

**Comment by qinenergy** [comment]
Hi! Glad you are able to reproduce our results now.

One of the main reason to prevent forgetting in continual test-time domain adaptation is that the learning in our setup is without labels. Therefore, errors from the pseudo-labels will accumulate and existing knowledge will be forgotten because of error accumulation. This phenomenon can also be observed in TENT https://github.com/DequanWang/tent/issues/7 as the performance drops if it revisits the same data again. It can also be seen quite obviously in Table 4 (where pseudo-label and TENT suffer more and more in later domains comparing to source-only). 

The forgetting from continual learning is a bit different as their setup is often supervised, and the forgetting comes from the sequential learning of tasks instead of error accumulation.

For the restoration factor choice, you can check the supplementary for an ablation and discussion.

To sum up, both continual test-time domain adaptation and continual learning suffer from forgetting. But the cause is a bit different. Test-time DA's forgetting is from error accumulation and CL's forgetting is from sequential learning. 

Thanks for trying our code and digging into it!

**Comment by kadmkbl** [comment]
Thank you so much for so nice and patient replies. I'll think it over by myself.
And congrats for acceptance of CoTTA :)
